src/kagglehub/config.py

```diff
@@ -144,3 +147,24 @@ def set_kaggle_credentials(username: str, api_key: str) -> None:
 def clear_kaggle_credentials() -> None:
     global _kaggle_credentials  # noqa: PLW0603
     _kaggle_credentials = None
+
+
+def get_colab_credentials() -> Optional[tuple[str, str]]:
+    # userdata access is already thread-safe after b/318732641
+    try:
+        from google.colab import userdata  # type: ignore[import]
+        from google.colab.errors import Error as ColabError  # type: ignore[import]
+    except ImportError:
+        return None
+
+    try:
+        username = userdata.get(COLAB_SECRET_USERNAME)
+        key = userdata.get(COLAB_SECRET_KEY)
+        return (username, key)
```

Comment:
Should you check that these are not empty?

Reply:
Additionally, Im not a big fan of hidden dependencies. Using namespaces or optional dependencies might be a good way to address this.

Reply:
Thanks @neshdev can you elaborate a little more what you mean?

Reply:
For emptiness test - Done!